third_party/abseil-cpp/absl/synchronization/mutex_starboard.cc

```diff
@@ -0,0 +1,87 @@
+// Copyright 2024 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "absl/synchronization/mutex.h"
+
+#include "starboard/mutex.h"
```

Comment:
You can use pthread_mutex_t as long as it is independent from other pthread objects.
We don't have the pthread_cond_t or pthread_once_t yet, but stand alone mutex use should work.

Reply:
Resolved offline, it seems the upstream code is doing something much more involved so we're going to go with the starboard APIs until we have broader pthread support